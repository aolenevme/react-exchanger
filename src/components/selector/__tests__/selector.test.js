import React from "react";
import renderer from "react-test-renderer";

import Selector from "../selector.js";

describe("<Selector />", () => {
    it("is rendered correctly", () => {
        expect(renderer
            .create(<Selector ratio={1.457} currencySigns={{
                fromCurrencySign: "£",
                toCurrencySign: "$"
            }}/>)
            .toJSON()).toMatchSnapshot();
    });

    it("is not rendered when props are incorrect", () => {
        expect(renderer
            .create(<Selector ratio="it-is-not-a-number" currencySigns={{
                fromCurrencySign: "£",
                toCurrencySign: "$"
            }}/>).toJSON())
            .toMatchSnapshot();

        expect(renderer
            .create(<Selector ratio={1.457} currencySigns={{
                fromCurrencySign: {not: "char"},
                toCurrencySign: "$"
            }}/>).toJSON())
            .toMatchSnapshot();

        expect(renderer
            .create(<Selector ratio={1.457} currencySigns={{
                fromCurrencySign: "£",
                toCurrencySign: {not: "char"}
            }}/>).toJSON())
            .toMatchSnapshot();
    });
});
