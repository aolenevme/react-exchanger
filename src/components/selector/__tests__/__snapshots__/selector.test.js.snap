// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Selector /> is not rendered when props are incorrect 1`] = `null`;

exports[`<Selector /> is not rendered when props are incorrect 2`] = `null`;

exports[`<Selector /> is not rendered when props are incorrect 3`] = `null`;

exports[`<Selector /> is rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.2rem 0.5rem;
  min-height: 2rem;
  border: 1px solid;
  border-radius: 4px;
  font-weight: 400;
  cursor: pointer;
  -webkit-transition: border-color 0.1s ease 0s;
  transition: border-color 0.1s ease 0s;
}

.c1 {
  width: 100%;
  text-align: center;
}

.c2 {
  font-size: 0.8rem;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    Â£1 = $1.45
    <span
      className="c2"
    >
      70
    </span>
  </span>
</div>
`;
